依赖于Qt 包的交叉编译存在问题

### Xmake 版本

3.0.8

### 操作系统版本和架构

ubuntu

### 描述问题

目前存在两个问题

# 第一
在PR(#7493)修复后，当一个包依赖于Qt时，在交叉编译的情况下，xmake 编译包时，不会传递qt相关的配置到包的config环节。（报错误 Qt SDK not found！）
编译应用本身而不是包，则会顺利编译通过。

以下是xmake -vD的部分输出，可以看到qt 和 qt_host 都没进入到包的配置的环节。
```
# 省略包的一些checking日志....
/usr/bin/xmake f --diagnosis --verbose -y -c --plat=cross --arch=arm64 --mode=debug --cross=aarch64-buildroot-linux-gnu- --bin=/home/mcgs/sdk/aic800-arm64-Qt5.15.15/bin --sdk=/home/mcgs/sdk/aic800-arm64-Qt5.15.15 --runtimes=MDd --toolchain=mcgs_aic800 --cxflags=-fPIC --builddir=build_e16bb86d --kind=shared
```

xmake配置如下
```
"xmake.additionalConfigArguments": [
        "--qt_host=/usr/local/MCGS/Qt5.15/5.15.2/gcc_64",
        "--qt=/home/mcgs/sdk/aic800-arm64-Qt5.15.15/bin",
        "--plat=cross",
        "--builddir=build_mcgs_aic800",
        "--toolchain=mcgs_aic800",
        "--arch=arm64"
]
```

# 第二
xmake的config会触发包的安装，但此时config并未完成，因此包的安装是根据当前检测环境进行的（而不是项目内的配置），而这通常编译安装的是不符合要求的。

这逻辑是否要修正一下？

### 期待的结果

希望依赖于Qt的包能够正常的编译。

第一个问题，这部分可以给我点指引，我去提个PR，因为包的安装逻辑还没搞懂xmake的源码在哪块。

第二个问题，目前是这么workaround
- 首先注释掉项目内的包的依赖
- 进行config
- 打开项目的包依赖
此后包的config环节就会和项目内一致了。但仍存在qt和qt_host未传递的问题。

### 工程配置

如描述

### 附加信息和错误日志

无

修改cross透传，如下，是可以进行编译了

<img width="1425" height="264" alt="Image" src="https://github.com/user-attachments/assets/6828fa64-a0bc-4d74-a84f-613410c47ceb" />

<img width="1440" height="769" alt="Image" src="https://github.com/user-attachments/assets/9e47d7a7-24a3-476a-a6b6-03da078ac240" />

依赖于Qt的包是编译通过，但是包本身进行链接时，出错了，额外链接了一个 `-lqtlibpng`，但这个ld找不到
```
/home/mcgs/sdk/aic800-arm64-Qt5.15.15/bin/../lib/gcc/aarch64-buildroot-linux-gnu/8.3.0/../../../../aarch64-buildroot-linux-gnu/bin/ld: cannot find -lqtlibpng
```

而使用同样的工具链和配置，编译应用 而不是包（两者都是依赖于qt，但是没有引入和png相关的内容），则不会引入这样的一个 `-lqtlibpng`
```
[ 59%]: <test> linking.debug test
/home/mcgs/sdk/aic800-arm64-Qt5.15.15/bin/aarch64-buildroot-linux-gnu-g++ -o build_mcgs_aic800/cross/arm64/debug/test build_mcgs_aic800/.objs/test/cross/arm64/debug/src/main.cpp.o build_mcgs_aic800/.objs/test/cross/arm64/debug/src/mainwindow.cpp.o build_mcgs_aic800/.objs/test/cross/arm64/debug/gens/src/moc_mainwindow.cpp.o --sysroot=/home/mcgs/sdk/aic800-arm64-Qt5.15.15/aarch64-buildroot-linux-gnu/sysroot -lQt5Gui -lQt5Widgets -lQt5Core
```
看起来是 xmake 做了个额外的静态链接库的查找，不清楚是不是这里引起的`rules/qt/load.lua`？

<img width="981" height="305" alt="Image" src="https://github.com/user-attachments/assets/12b4d2b2-ae7e-4b99-847c-ebc168d670df" />

--- a/xmake/modules/detect/sdks/find_qt.lua
+++ b/xmake/modules/detect/sdks/find_qt.lua
@@ -365,10 +365,10 @@ function main(sdkdir, opt)
     end
 
     -- find qt
-    local sdkdir = sdkdir or config.get("qt") or global.get("qt") or config.get("sdk")
+    local real_sdkdir = sdkdir or config.get("qt") or global.get("qt") or config.get("sdk")
     local sdkver = opt.version or config.get("qt_sdkver")
     local sdkdir_host = opt.sdkdir_host or config.get("qt_host") or global.get("qt_host")
-    local qt = _find_qt(sdkdir, sdkver, sdkdir_host)
+    local qt = _find_qt(real_sdkdir, sdkver, sdkdir_host)
     if qt then
         config.set("qt", qt.sdkdir, {force = true, readonly = true})
         config.set("qt_sdkver", qt.sdkver, {force = true, readonly = true})

--- a/xmake/modules/package/tools/xmake.lua
+++ b/xmake/modules/package/tools/xmake.lua
@@ -49,7 +49,7 @@ end
 
 -- get configs for qt
 function _get_configs_for_qt(package, configs, opt)
-    local names = {"qt", "qt_sdkver"}
+    local names = {"qt", "qt_sdkver", "qt_host"}
     for _, name in ipairs(names) do
         local value = get_config(name)
         if value ~= nil then
@@ -226,6 +226,8 @@ function _get_configs_for_cross(package, configs, opt)
             table.insert(configs, "--" .. name .. "=" .. tostring(value))
         end
     end
+    _get_configs_for_qt(package, configs, opt)
+    _get_configs_for_vcpkg(package, configs, opt)
 end
 
 -- get configs

